gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/BigQueryImplTest.java

```diff
@@ -0,0 +1,982 @@
+    String cursor = "cursor";
+    ImmutableList<BaseTableInfo> tableList = ImmutableList.of(TABLE_INFO_WITH_PROJECT,
+        OTHER_TABLE_INFO);
+    Tuple<String, Iterable<Table>> result =
+        Tuple.of(cursor, Iterables.transform(tableList, BaseTableInfo.TO_PB_FUNCTION));
+    EasyMock.expect(bigqueryRpcMock.listTables(DATASET, TABLE_LIST_OPTIONS)).andReturn(result);
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    Page<BaseTableInfo> page = bigquery.listTables(DATASET, TABLE_LIST_MAX_RESULTS,
+        TABLE_LIST_PAGE_TOKEN);
+    assertEquals(cursor, page.nextPageCursor());
+    assertArrayEquals(tableList.toArray(), Iterables.toArray(page.values(), BaseTableInfo.class));
+  }
+
+  @Test
+  public void testDeleteTable() {
+    EasyMock.expect(bigqueryRpcMock.deleteTable(DATASET, TABLE)).andReturn(true);
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    assertTrue(bigquery.delete(DATASET, TABLE));
+  }
+
+  @Test
+  public void testDeleteTableFromTableId() {
+    EasyMock.expect(bigqueryRpcMock.deleteTable(DATASET, TABLE)).andReturn(true);
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    assertTrue(bigquery.delete(TABLE_ID));
+  }
+
+  @Test
+  public void testUpdateTable() {
+    BaseTableInfo updatedTableInfo = TABLE_INFO.toBuilder().description("newDescription").build();
+    BaseTableInfo updatedTableInfoWithProject = TABLE_INFO_WITH_PROJECT.toBuilder()
+        .description("newDescription")
+        .build();
+    EasyMock.expect(bigqueryRpcMock.patch(updatedTableInfoWithProject.toPb(), EMPTY_RPC_OPTIONS))
+        .andReturn(updatedTableInfoWithProject.toPb());
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    BaseTableInfo table = bigquery.update(updatedTableInfo);
+    assertEquals(updatedTableInfoWithProject, table);
+  }
+
+  @Test
+  public void testUpdateTableWithSelectedFields() {
+    Capture<Map<BigQueryRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    BaseTableInfo updatedTableInfo = TABLE_INFO.toBuilder().description("newDescription").build();
+    BaseTableInfo updatedTableInfoWithProject = TABLE_INFO_WITH_PROJECT.toBuilder()
+        .description("newDescription")
+        .build();
+    EasyMock.expect(bigqueryRpcMock.patch(eq(updatedTableInfoWithProject.toPb()),
+        capture(capturedOptions))).andReturn(updatedTableInfoWithProject.toPb());
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    BaseTableInfo table = bigquery.update(updatedTableInfo, TABLE_OPTION_FIELDS);
+    String selector = (String) capturedOptions.getValue().get(TABLE_OPTION_FIELDS.rpcOption());
+    assertTrue(selector.contains("tableReference"));
+    assertTrue(selector.contains("schema"));
+    assertTrue(selector.contains("etag"));
+    assertEquals(31, selector.length());
+    assertEquals(updatedTableInfoWithProject, table);
+  }
+
+  @Test
+  public void testInsertAll() {
```

Comment:
Do you think it'd be worthwhile to test that `skipInvalidRows` and `ignoreUnknownValues` are being handled properly in this method?